spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/launcher/TaskLauncherSink.java

```diff
@@ -48,19 +53,48 @@
 	@Autowired
 	private DelegatingResourceLoader resourceLoader;
 
+	private ObjectMapper objectMapper;
+
+
+	public TaskLauncherSink() {
+		this.objectMapper = new ObjectMapper();
+	}
+
 	/**
 	 * Launches a task upon the receipt of a valid TaskLaunchRequest.
 	 * @param request is a TaskLaunchRequest containing the information required to launch
 	 * a task.
 	 */
 	@ServiceActivator(inputChannel = Sink.INPUT)
-	public void taskLauncherSink(TaskLaunchRequest request) {
-		launchTask(request);
+	public void taskLauncherSink(Message<?> request) throws Exception{
+		TaskLaunchRequest taskLaunchRequest = null;
+		if(request.getPayload() instanceof String) {
+			taskLaunchRequest = this.objectMapper.readValue(((String)request.getPayload()), TaskLaunchRequest.class);
+		}
+		else if(request.getPayload() instanceof byte[]) {
+			taskLaunchRequest = this.objectMapper.readValue(((byte[])request.getPayload()), TaskLaunchRequest.class);
+		}
+		else if(request.getPayload() instanceof TaskLaunchRequest) {
+			taskLaunchRequest = (TaskLaunchRequest)request.getPayload();
+		}
+		else if(request.getPayload() instanceof Map) {
+			Map<String, Object> linkedHashMap = (Map<String, Object>)request.getPayload();
+			taskLaunchRequest = new TaskLaunchRequest((String)linkedHashMap.get("uri"),
+					(List<String>) linkedHashMap.get("commandlineArguments"),
+					(Map)linkedHashMap.get("environmentProperties"),
+					(Map)linkedHashMap.get("deploymentProperties"),
+					(String)linkedHashMap.get("applicationName"));
+		}
+		else {
+			throw new IllegalArgumentException("Unsupported payload type");
+		}
```

Comment:
Why do we need to do this type conversion in our code and not have SCSt handle it?

Reply:
Adding @sobychacko for color commentary.  There maybe a better way to handle this, but it was my understanding that this was the correct method.

Reply:
cc: @olegz Reading this code makes me feel like either we're doing it wrong or there is something missing from Spring Cloud Stream.

Reply:
Let me add a little more commentary.   We looked at using the https://github.com/spring-cloud-stream-app-starters/core/blob/master/common/app-starters-postprocessor-common/src/main/java/org/springframework/cloud/stream/app/postprocessor/ContentTypeEnvironmentPostProcessor.java[ContentTypeEnvironmentPostProcessor] available in the Spring Cloud Stream Apps Starters project, but that didn't translate well for SCT.  We could have implemented for the TaskLauncherStarter, but in this case the solution work for those who wish to use the TaskLauncher outside of the starters project.

Reply:
@cppwfs I might have missed this context when we chatted. However, since you made the `TaskLaunchRequest` object `json` friendly, I think you can simply rely on the default content type used by spring cloud stream which is `application/json` to do the conversion. It should convert the payload appropriately to the POJO and you can remove all the object mapper based conversions as the the incoming payload is already `TaskLaunchRequest`.

Reply:
You can force it to use `application/json` as the content type by using this property: `spring.cloud.stream.bindings.input.contentType=application/json`